app/Services/VCard/ExportVCard.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace App\Services\VCard;
+
+use App\Services\BaseService;
+use App\Models\Contact\Gender;
+use App\Models\Contact\Contact;
+use Sabre\VObject\Component\VCard;
+
+class ExportVCard extends BaseService
+{
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'contact_id' => 'nullable|integer',
+        ];
+    }
+
+    /**
+     * Import one VCard.
+     *
+     * @param array $data
+     * @return VCard
+     */
+    public function execute(array $data) : VCard
+    {
+        $this->validate($data);
+
+        $contact = Contact::where('account_id', $data['account_id'])
+            ->findOrFail($data['contact_id']);
+
+        return $this->export($contact);
+    }
+
+    private function escape($value) : string
+    {
+        return ! empty((string) $value) ? trim((string) $value) : (string) null;
+    }
+
+    /**
+     * @param Contact $contact
+     * @return VCard
+     */
+    private function export(Contact $contact) : VCard
+    {
+        // The standard for most of these fields can be found on https://tools.ietf.org/html/rfc6350
+
+        // Basic information
+        $vcard = new VCard([
+            'UID' => $contact->hashid(),
+            'SOURCE' => route('people.show', $contact),
+        ]);
+
+        $this->exportNames($contact, $vcard);
```

Comment:
Instead of repeating those two attributes everywhere, isn't it simpler to create  Contact and vCard objects at the class level at the beginning of this file? Like `private Contact $contact` and tell call `$this->contact`?

Reply:
Well, it's more thread safe like this, but I agree there are other solutions.
But like this, it's way way easier to test it!